src/pages/experiments/[experimentId]/data-processing/index.jsx

```diff
@@ -91,34 +91,42 @@ const DataProcessingPage = ({ experimentId, experimentData, route }) => {
   }, [experimentId]);
 
   useEffect(() => {
-    if (samples.meta.loading) dispatch(loadSamples(experimentId))
-  }, [samples.meta.loading])
+    if (samples.meta.loading) {
+      dispatch(loadSamples(experimentId));
+      return;
+    }
 
-  useEffect(() => {
-    if (samples.ids.length) {
+    if (samples.ids.length > 0) {
       setPreFilteredSamples(
         samples.ids.reduce(
           (acc, sampleUuid) => samples[sampleUuid].preFiltered ? [...acc, sampleUuid] : acc, []
         )
       )
     }
-  }, [samples])
 
-  useEffect(() => {
+  }, [samples.meta.loading])
 
-    if (preFilteredSamples.length && processingConfig[steps[stepIdx].key].enabled) {
```

Comment:
The issue was `processingConfig[steps[stepIdx].key]` was not defined at the point of this hook launching. We did not check this and because processing config was not semaphored to this hook, we would not have been able to detect changes anyways. This did not pose a problem locally because `preFilteredSamples` was always empty, so the code never progressed to the problematic second condition.